src/main/java/io/github/ascopes/protobufmavenplugin/AbstractGenerateMojo.java

```diff
@@ -480,6 +480,12 @@ public abstract class AbstractGenerateMojo extends AbstractMojo {
   @Parameter(defaultValue = "false")
   boolean rustEnabled;
 
+  @Parameter(defaultValue = "false")
+  boolean pluginOnly;
+
+  @Parameter
+  @Nullable String pluginOption;
```

Comment:
Would it be possible to add a JavaDoc to this field? Reason being it will be used to generate the documentation for this parameter in the generated GitHub pages. Worth adding a `@since 1.3.0` here as well.

Reply:
I am wondering if we want to include metadata on plugins in a more controlled way whether it is worth revamping how plugins are defined so that you can tie options down to the specific plugins they apply to.

That'd be an implication of a v2.0.0 release due to the breaking change, but I am more than happy to explore that possibility.

I did look into the idea of having just one way of specifying plugins that let you specify one of the path name, the URL, or the artifact... but I couldn't think of a good typesafe way of doing it. Nothing stopping me making another attempt though.

```
<plugins>
  <plugin>
    <type>PATH</type>
    <name>protoc-gen-grpc-java</name>
    <options>
      <option>foobarbaz</option>
    </options>
  </plugin>
</plugins>
```

What are your thoughts on this?